pkg/handlers/v1/eni.go

```diff
@@ -69,7 +82,38 @@ func (t eniTransformer) Update(event awsConfigEvent) (Output, bool, error) {
 		return Output{}, false, err
 	}
 
-	return output, filter(config), nil
+	//return output, filter(config), nil
+
+	addedChange := Change{ChangeType: added}
+	deletedChange := Change{ChangeType: deleted}
+	// If an update was detected, check to see if any changes to the NetworkInterfaces occurred
+	for k, v := range event.ConfigurationItemDiff.ChangedProperties {
+		if !strings.HasPrefix(k, "Configuration.PrivateIpAddresses.") {
+			continue
+		}
+		var diff PrivateIPBlockDiff
+		if err := json.Unmarshal(v, &diff); err != nil {
+			return Output{}, false, err
+		}
+		ipBlock := diff.UpdatedValue
+		changes := &addedChange
+		if diff.ChangeType == delete {
+			ipBlock = diff.PreviousValue
+			changes = &deletedChange
+		}
+		extractIPBlock(ipBlock, changes)
+		extractRelatedResources(&config, changes)
+	}
+	// We need to compute the symmetric difference of the added changes and the removed changes
+	// i.e. remove entries that show up as both added and removed
+	symmetricDifference(&addedChange, &deletedChange)
```

Comment:
this symmetricDifference function is already defined in the ec2.go file, which is what this file references. I was wondering if I could get some insight on a better place to declare that function so that it makes more sense.

Reply:
I'm open to it if that's the route we want to go. Speaking specifically to the symmetricDifference function, moving this would be easy because the two files are using the exact same function. However, for the other shared code, I saw some nuances that make them slightly different (e.g. sending parameters into a function to do the logic on vs defining them in the function itself and then doing the logic). I'm worried if moving over those types of functions could result in unexpected errors that would take more than a simple refactor.